### ISSUE TITLE
uwsgi conf to isolate an app using buildout

### ISSUE THREAD

**Issue by sandroden** [buildout/buildout#446]
Hi,

i recently realized that my buildout projects are not really isolated from system python when i deploy them using uwsgi.

In my specific case I deploy django project and buildout correctly builds bin/django.wsgi as a script that has all the path correctly set. I reference this file in the conf  of uwsgi:

```
env = DJANGO_SETTINGS_MODULE=web.settings
prj_dir = /home/www/3palchi-test
wsgi-file = %(prj_dir)/bin/django.wsgi
```
And I use uwsgi-emperor configuration. This configuration works pretty well but really it is not isolated and as uwsgi interpreter adds 

```
/usr/lib/python2.7
/usr/lib/python2.7/plat-x86_64-linux-gnu
/usr/lib/python2.7/lib-tk
/usr/lib/python2.7/lib-old
/usr/lib/python2.7/lib-dynload
/usr/local/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages

```
Last week, a packages with a namespace that has some problems on it's own (but working regularly) broke the import. 

How do you folk configure it to isolate it? Note that I always start from a "clean" or "empty" virtualenv that I call base2 for py2 and base3 for py3. I use it's python interpreter to bootstrap the setup:
`
/home/www/.virtualenvs/base/bin/python2  bootstrap.py`

so I cannot declare base2 as the virtualenv directive in the uwsgi conf file, as this is really a virtualenv that is missing all the packages....

**Comment by jimfulton** [comment]
I vaguely remember having to fight this a few years ago using uwsgi....

How is uwsgi installed? Is it installed by buildout? Part of your virtualenv?

**Comment by reinout** [comment]
It isn't installed by buildout, but separately. See this similar issue in djangorecipe: https://github.com/rvanlaar/djangorecipe/issues/98

I don't know how uwsgi works, but if it is started outside of buildout, starting a wsgi file that buildout/djangorecipe generates probably cannot "fix" the `sys.path` anymore?

The simplest solution seems to me to install uwsgi inside the buildout. That ought to fix everything.

**Comment by sandroden** [comment]
Adding the `virtualenv` directive in uwsgi solved the issue. I did try it before, but looking it back I realized I added a typo (`/var` instead of `/home`) so that I complained it didn't fine `site`. Thanks to Riccardo Magliocchetti on uwsgi issue that forced me to re-check.   

Just for completeness I add the (part of the) configuration that is working for me:

```
 [uwsgi]

env = DJANGO_SETTINGS_MODULE=web.settings
env = HOME=/tmp
env = LC_ALL=it_IT.UTF-8
prj_dir = /home/www/3palchi
wsgi-file = %(prj_dir)/bin/django.wsgi
virtualenv = /home/www/.virtualenvs/base2
```
